src/core/lib/event_engine/thread_pool/work_stealing_thread_pool.cc

```diff
@@ -55,7 +55,9 @@ constexpr grpc_core::Duration kLifeguardMinSleepBetweenChecks{
     grpc_core::Duration::Milliseconds(15)};
 constexpr grpc_core::Duration kLifeguardMaxSleepBetweenChecks{
     grpc_core::Duration::Seconds(1)};
-constexpr absl::Duration kSleepBetweenQuiesceCheck{absl::Milliseconds(10)};
+// TODO(hork): quiesce may be faster and more efficient with a lock + condition
+// on thread_count = N.
+constexpr absl::Duration kSleepBetweenQuiesceCheck{absl::Microseconds(50)};
```

Comment:
This gets close to busy waiting. Is the Quiesce time crucial for performance ?

Reply:
> This gets close to busy waiting.

Exactly, that's the reasoning behind the TODO. Introducing a new mutex + condvar wait could eliminate the loop, but quiesce should be a fairly infrequent thing, and will hopefully not take long when it does happen.

>  Is the Quiesce time crucial for performance?

It makes stress testing feasible.

Reply:
I removed the busy loop. PTAL